### PR TITLE
feat: Add app fetching

### PR DESCRIPTION
**Pull request by ptbrowne**
Add app fetching and manifest sanitization.

### CODE REVIEW

File: packages/cozy-client/src/manifest.js [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+  const filteredList = categories.filter(c => AUTHORIZED_CATEGORIES.includes(c))
+  if (!filteredList.length) return ['others']
+  return filteredList
+}
+
+export function areTermsValid(terms) {
+  return !!terms && !!terms.id && !!terms.url && !!terms.version
+}
+
+export function isParnershipValid(partnership) {
```
Comment (drazik) [inline packages/cozy-client/src/manifest.js]: `isPartnershipValid` (a `t` is missing)

File: packages/cozy-client/src/manifest.js [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+  return !!partnership && !!partnership.description
```
Comment (drazik) [inline packages/cozy-client/src/manifest.js]: Nit: I think `Boolean(partnership && partnership.description)` is more readable if you absolutely want to return a real boolean.

File: packages/cozy-client/src/manifest.js [hunk 3]
```diff
@@ -0,0 +1,85 @@
+const AUTHORIZED_CATEGORIES = [
```
Comment (edas) [inline packages/cozy-client/src/manifest.js]: categories of what ? I do not see that self-explainatory. Maybe APP_CATEGORY (and same thing for the function below).

+ I would externalize the list of categories in a json or in its own JS files

File: packages/cozy-client/src/manifest.js [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+  'public_service',
+  'shopping',
+  'social',
+  'telecom',
+  'transport'
+]
+
+/** Filters unauthorized categories. Defaults to ['others'] if no suitable category. */
+export function sanitizeCategories(categories) {
+  if (!categories) return ['others']
```
Comment (edas) [inline packages/cozy-client/src/manifest.js]: strange as "others" is not a valid category

File: packages/cozy-client/src/manifest.js [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+ *
+ * @param  {Manifest} manifest
+ * @return {Manifest}
+ */
+export function sanitize(manifest) {
+  const sanitized = { ...manifest }
+
+  // Make categories an array and delete category attribute if it exists
+  if (
+    !manifest.categories &&
```
Comment (edas) [inline packages/cozy-client/src/manifest.js]: It's probably not for this PR but I really dislike these options that we have to maintain a long time. Should we create an issue about deprecating the  `category` option and only leave `categories`?

Comment (ptbrowne) [reply packages/cozy-client/src/manifest.js]: I do not know the status of the category option, I merely moved the code from the store here.

I have no strong opinion on the deprecation here, I do not know for how long the category option has been removed.

File: packages/cozy-client/src/manifest.js [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+    delete sanitized.terms
+  }
+
+  // Remove invalid partnership
+  if (sanitized.partnership && !isPartnershipValid(sanitized.partnership)) {
+    delete sanitized.partnership
+  }
+
+  return sanitized
+}
```
Comment (Crash--) [inline packages/cozy-client/src/manifest.js:85]: I didn't even know about this file. I think we should move it to `/models` directory. Can you move it and create a BC for that?

**Pull request merged**